bin/worker.js

```diff
@@ -176,41 +172,35 @@ async function main () {
   // level docker-worker components.
   config.docker = require('../lib/docker')();
 
-  // Wrapped stats helper to support generators, etc...
-  config.stats = new Stats(config);
-  config.stats.record('workerStart', Date.now()-os.uptime() * 1000);
-  debug('passed stats')
-
-  config.queue = new taskcluster.Queue({
+  let monitor = await base.monitor({
+    project: 'docker-worker',
     credentials: config.taskcluster,
-    stats: base.stats.createAPIClientStatsHandler({
-      drain: config.stats.influx,
-      tags: {
-        component: 'docker-worker',
-        workerId: config.workerId,
-        workerGroup: config.workerGroup,
-        workerType: config.workerType,
-        instanceType: config.workerNodeType,
-        provisionerId: config.provisionerId
-      }
-    })
+    mock: profile === 'test'
   });
 
-  config.validator = await base.validator();
-  config.validator.register(require('../schemas/payload'));
+  config.monitor = monitor.prefix(
+    `${config.provisionerId}.` +
+    `${config.workerGroup}.${config.workerType}.` +
+    `${config.workerNodeType.replace('.', '')}`
+  );
 
-  // Only catch these metrics when running on aws host.  Running within
-  // test environment causes numerous issues.
-  if (host === 'aws') {
-    base.stats.startProcessUsageReporting({
-      drain: config.stats.influx,
-      component: 'docker-worker',
-      process: 'docker-worker'
-    });
-  }
+  config.monitor.measure('workerStart', Date.now()-os.uptime());
+  config.monitor.count('workerStart');
+
+  config.queue = new taskcluster.Queue({
+    credentials: config.taskcluster
+  });
+
+  config.validator = await base.validator({
+    folder: path.join(__dirname, '..', 'schemas'),
```

Comment:
`base.validator` _should_ handle this correctly by default I think. If it's not working for you without this it might be considered a bug on `tc-lib-validator`.